src/Cleanup.php

```diff
@@ -89,7 +89,32 @@ public function preMigration(string $mainRoleName): void
     {
         $sqls = [];
         $currentRole = $this->config->getTargetSnowflakeRole();
-        $this->destinationConnection->grantRoleToUser($this->config->getTargetSnowflakeUser(), $mainRoleName);
+
+        $mainRole = $this->destinationConnection->fetchAll(sprintf(
+            'SHOW ROLES LIKE %s',
+            QueryBuilder::quote($mainRoleName),
+        ));
+        $grantsToTargetUser = $this->destinationConnection->fetchAll(sprintf(
+            'SHOW GRANTS TO USER %s',
+            QueryBuilder::quoteIdentifier($this->config->getTargetSnowflakeUser()),
+        ));
+        $mainRoleExistsOnTargetUser = array_filter(
+            $grantsToTargetUser,
+            fn($v) => $v['role'] === $mainRoleName,
+        );
```

Comment:
Když to reprezentuje true/false, tak co raději

```suggestion
        $mainRoleExistsOnTargetUser = array_reduce(
            $grantsToTargetUser,
            fn ($found, $v) => $found || $v['role'] === $mainRoleName,
            false,
        );
```